app/javascript/src/Dashboard/components/BackendsWidget.jsx

```diff
@@ -21,10 +43,114 @@ type Props = {
 }
 
 const BackendsWidget = (props: Props) => {
-  console.log(props)
+  console.log('what are the props' + JSON.stringify(props))
+  console.log('what are the props 2' + JSON.stringify(props.backends[0].name))
+
+  const [ isOpenArray, setIsOpenArray ] = useState([])
+
+  function handleChange (e) {
+    const item = e.target.id
+    var indexOfItem = isOpenArray.indexOf(item)
+
+    if (indexOfItem !== -1) {
+      var array = [...isOpenArray]
+      if (indexOfItem === 0) {
+        array.shift()
+        setIsOpenArray(array)
+      } else {
+        var newArray = array.splice(indexOfItem, 1)
+        setIsOpenArray(newArray)
+      }
+    } else {
+      setIsOpenArray([item, ...isOpenArray])
+    }
+  }
+
+  const APIDataListItem = props.backends.map((api, index) => {
+    let dateUpdatedAt = new Date(api.updated_at)
+
+    return (
+      <DataListItem aria-labelledby="single-action-item1">
+        <DataListItemRow>
+          <DataListItemCells
+            dataListCells={[
+              <DataListCell key="primary content">
+                <a href={api.link} id="single-action-item1">
+                  {api.name}
+                </a>
+              </DataListCell>,
+              <DataListCell key="secondary content" className="dashboard-list-secondary">
+                {dateUpdatedAt.toUTCString()}
+              </DataListCell>
+            ]}
+          />
+          <DataListAction
+            aria-labelledby="multi-actions-item1 multi-actions-action1"
+            id="actions-menu"
+            aria-label="Actions"
+            isPlainButtonAction
+          >
+            <Dropdown
+              isPlain
+              id="actions-menu"
+              position={DropdownPosition.right}
+              isOpen={isOpenArray.indexOf(`toggle-${index}`) !== -1}
```

Comment:
Instead of `isOpenArray`, I think it would be simpler to extract a custom Dropdown with its own `isOpen` state and handler.

Reply:
could you show an example of this ? i struggled working with the patternfly component

Reply:
Sure thing, I'll push it in a branch to your repo!